research/api-comparision.md

```diff
@@ -0,0 +1,618 @@
+
+/**
+* LaunchDarkly
+*
+* SDK Repo: https://github.com/launchdarkly/java-server-sdk
+*/
+boolean boolVariation(String featureKey, LDUser user, boolean defaultValue)
+// Response also contains evaluation details
+EvaluationDetail<Boolean> boolVariationDetail(String featureKey, LDUser user, boolean defaultValue)
+
+/**
+* Split
+*
+* SDK Repo: https://github.com/splitio/java-client
+*
+* NOTE: Not supported
+* NOTE: Split always returns a string but using the values "on" and "off" is a common practice.
+*/
+
+/**
+* CloudBees Rollout
+*
+* SDK Repo: N/A
+*
+* NOTE: Flags are configured as code and contain default values.
+*
+*/
+public class Flags implements RoxContainer {
+  public RoxFlag videoChat = new RoxFlag();
+}
+Flags flags = new Flags();
+Rox.register("test-namespace", flags);
+flags.videoChat.enabled
+
+// Dynamic API
+Rox.dynamicApi.isEnabled('system.reportAnalytics', false);
+Rox.dynamicApi.isEnabled('system.reportAnalytics', false, context);
+
+/**
+* Harness
+*
+* SDK Repo: https://github.com/harness/ff-java-server-sdk
+*/
+boolean boolVariation(String key, Target target, boolean defaultValue)
+```
+
+</details>
+
+<details>
+  <summary>NodeJS</summary>
+
+```typescript
+/**
+* Unleash
+*
+* SDK Repo: https://github.com/Unleash/unleash-client-node
+*/
+isEnabled(name: string, context?: Context, fallbackFunction?: FallbackFunction): boolean;
+isEnabled(name: string, context?: Context, fallbackValue?: boolean): boolean;
+isEnabled(name: string, context: Context = {}, fallback?: FallbackFunction | boolean): boolean;
+
+/**
+* Flagsmith
+*
+* SDK Repo: https://github.com/Flagsmith/flagsmith-nodejs-client
+*/
+hasFeature(key: string): Promise<boolean>
+hasFeature(key: string, userId: string): Promise<boolean>
+
+/**
+* LaunchDarkly
+*
+* SDK Repo: https://github.com/launchdarkly/node-client-sdk
+*
+* Note: TS typings assign LDFlagValue = any;
+* Note: variation values can be defined as boolean, number, or string. Documentation suggests casting: https://docs.launchdarkly.com/sdk/client-side/node-js#getting-started
+*/
+
+variation(
+key: string,
+  user: LDUser,
+  defaultValue: LDFlagValue,
+  callback?: (err: any, res: LDFlagValue) => void
+): Promise<LDFlagValue>
+// Response also contains evaluation details
+variationDetail(
+  key: string,
+  user: LDUser,
+  defaultValue: LDFlagValue,
+  callback?: (err: any, res: LDEvaluationDetail) => void
+): Promise<LDEvaluationDetail>;
+
+/**
+* Split
+*
+* SDK Repo: https://github.com/splitio/javascript-client
+*
+* NOTE: Not supported
+* NOTE: TS typings assign Treatment = string;
+* NOTE: Split always returns a string but using the values "on" and "off" is a common practice.
+*/
+
+/**
+* CloudBees Rollout
+*
+* SDK Repo: N/A
+*
+* NOTE: Flags are configured as code and contain default values.
+*
+*/
+const flags = {
+  videoChat: new Rox.Flag()
+};
+
+Rox.register('test-namespace', flags);
+flags.videoChat.isEnabled()
+flags.videoChat.isEnabled(context)
+
+export class Flag {
```

Comment:
How is this class used?

Reply:
That's just the instance that's created when you define a boolean flag object. It's just the class that houses the `getValue()` method.

Reply:
I've removed the class context, it's not needed, the Java sections only have method signatures... and if the classes aren't needed in the Java samples( the "classiest" language there is) we don't need them in TS.